### PR TITLE
DM-12635: Gen3 Butler specializations

### PR DESCRIPTION
**Pull request by TallJimbo**
Note that this PR is relative to the new `gen3-middleware` integration branch, not master.

### CODE REVIEW

File: python/lsst/obs/subaru/gen3.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    def __init__(self, mapper):
+        self.sensors = [
+            dict(
+                sensor=sensor.getId(),
+                name=sensor.getName(),
+                # getType() returns a pybind11-wrapped enum, which
+                # unfortunately has no way to extract the name of just
+                # the value (it's always prefixed by the enum type name).
+                purpose=str(sensor.getType()).split(".")[-1],
```
Comment (pschella) [inline python/lsst/obs/subaru/gen3.py]: Have you tried adding `py::arithmetic()` to the `enum` wrapper? That gives you conversion to value too I think.

File: python/lsst/obs/subaru/gen3.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            )
+
+
+class HscAbstractFilterKeyHandler(KeyHandler):
+    """KeyHandler for HSC filter keys that should be mapped to AbstractFilters.
+    """
+
+    __slots__ = ()
+
+    REGEX = re.compile(r"HSC\-([GRIZY])2?")
```
Comment (pschella) [inline python/lsst/obs/subaru/gen3.py]: Why compile it here but not above?

File: python/lsst/obs/subaru/gen3.py (hunk 1 above)
Comment (TallJimbo) [reply python/lsst/obs/subaru/gen3.py]: I'll open another ticket for that; I'd rather not have to make afw changes for this one.

File: python/lsst/obs/subaru/gen3.py (hunk 2 above)
Comment (TallJimbo) [reply python/lsst/obs/subaru/gen3.py]: Heh, just wrote them at completely different times.  Will fix.

File: python/lsst/obs/subaru/gen3.py (hunk 1 above)
Comment (TallJimbo) [reply python/lsst/obs/subaru/gen3.py]: https://jira.lsstcorp.org/browse/DM-14247

**Pull request merged**